app/api/helpers/permission_manager.py

```diff
@@ -129,6 +130,11 @@ def is_coorganizer_or_user_itself(view, view_args, view_kwargs, *args, **kwargs)
     if user.is_organizer(kwargs['event_id']) or user.is_coorganizer(kwargs['event_id']):
         return view(*view_args, **view_kwargs)
 
+    if ('model' in kwargs) and (kwargs['model'] == Speaker):
```

Comment:
Why not create a separate decorator?

Reply:
@iamareebjamal Since it was being used only in orders and speakers at Resource Details each only, I thought I would just add a condition here, if its better, I can create a separate decorator altogether!

Reply:
Yes create a separate one

Reply:
Understood, shall do it!